### PR TITLE
Fix Mixing hparams and arguments in LightningModule

### LINKED ISSUE #1468: Mixing hparams and arguments in LightningModule.__init__() crashes load_from_checkpoint()
## 🐛 Bug

Right now, if you initialize a Lightning Module with a mixture of a Namespace (hparams) as well as additional arguments (say to a Dataset), load_from_checkpoint can't recover.

### To Reproduce

Create a LightningModule as follows:

```python
class Model(pl.LightningModule):
      def __init__(self, hparams, train_dataset, val_dataset):
              self.hparams = hparams
              self.train_dset, self.val_dset = train_dataset, val_dataset
              ...
```

Run training, then try to restore from checkpoint, via:

```python
nn = Model.restore_from_checkpoint(<PATH>, train_dataset=None, val_dataset=None)
```

### Expected behavior

Ideally, you'd just be able to pass in the additional arguments (as above) and everything would work.

### PR DESCRIPTION
**Pull request by HenryJia**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
Fixes #1468 

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by HenryJia** [comment]
As in allow things in kwargs to override model_args?

**Comment by williamFalcon** [comment]
i say keep it simple as is now

**Comment by williamFalcon** [comment]
@HenryJia can you update the docs to show this?

**Comment by HenryJia** [comment]
@williamFalcon Yep, done

**Comment by HenryJia** [comment]
@Borda Added note to changelog, looking into clean ways of adding a test for this...

**Comment by awaelchli** [comment]
The note you added in the documentation could be confusing. I wouldn't call these extra arguments hyperparameters. Because to me it reads like you can do `load_from_checkpoint(file, learning_rate=2)` and then magically somewhere in Lightningmodule access it with hparams like `self.hparams.learning_rate`. But clearly this is not what you try to achieve. You simply want to pass in some extra args, in addition to hparams dict. Apart from that, I don't see why this implementation prevents anyone passing in arguments that ARE part of the hparams dict. 

A reformulation and an example in the docs would make it clear (at least for me).

**Comment by HenryJia** [comment]
@awaelchli Sounds good I'll change it slightly then. Regarding the example, htere already is one in the docs. It's the last example of that section.

**Comment by awaelchli** [comment]
Awesome, it's cristal clear now. Thanks for the fix.

File: CHANGELOG.md [hunk 1]
```diff
@@ -39,9 +39,12 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Fixed loggers - flushing last logged metrics even before continue, e.g. `trainer.test()` results ([#1459](https://github.com/PyTorchLightning/pytorch-lightning/pull/1459))
 
+- Fixed LightningModule - Mixing hparams and arguments in `LightningModule.__init__()` crashes load_from_checkpoint() ([#1505](https://github.com/PyTorchLightning/pytorch-lightning/pull/1505))
+
 - Added a missing call to the `on_before_zero_grad` model hook ([#1493](https://github.com/PyTorchLightning/pytorch-lightning/pull/1493)). 
 
 - 
+>>>>>>> master
```
Comment (Borda) [inline CHANGELOG.md:47]:
```suggestion
```

**Pull request merged**